libcontainer/cgroups/utils.go

```diff
@@ -415,23 +415,23 @@ func ConvertCPUSharesToCgroupV2Value(cpuShares uint64) uint64 {
 
 // ConvertMemorySwapToCgroupV2Value converts MemorySwap value from OCI spec
 // for use by cgroup v2 drivers. A conversion is needed since Resources.MemorySwap
-// is defined as memory+swap combined, while in cgroup v2 swap is a separate value.
+// is defined as memory+swap combined, while in cgroup v2 swap is a separate value,
+// so we need to subtract memory from it where it makes sense.
 func ConvertMemorySwapToCgroupV2Value(memorySwap, memory int64) (int64, error) {
-	// for compatibility with cgroup1 controller, set swap to unlimited in
+	// For compatibility with cgroup1 controller, set swap to unlimited in
 	// case the memory is set to unlimited, and swap is not explicitly set,
 	// treating the request as "set both memory and swap to unlimited".
 	if memory == -1 && memorySwap == 0 {
 		return -1, nil
 	}
-	if memorySwap == -1 || memorySwap == 0 {
-		// -1 is "max", 0 is "unset", so treat as is
+	if memorySwap == -1 || memorySwap == 0 || // Treat -1 ("max") and 0 ("unset") swap as is.
```

Comment:
any reason to not put this comment prior to the if?

Reply:
Because it is now related to this exact line, not the whole "if" statement. See:

```go
if memorySwap == -1 || memorySwap == 0 || // Treat -1 ("max") and 0 ("unset") swap as is.
	memory == -1 { // Unlimited memory, so treat swap as is.
	return memorySwap, nil
}
```
Basically, this is a shortened form of

```go
// Treat -1 ("max") and 0 ("unset") swap as is.
if memorySwap == -1 || memorySwap == 0 
	return memorySwap, nil
}

// Unlimited memory, so treat swap as is.
if memory == -1 {
	return memorySwap, nil
}
```

Let me know if you like the last version better and I will change it.

Reply:
I was thinking one comment for both statements is clear enough. But sure, the version with two ifs looks quite clear, if you don't mind :)

Reply:
@rata I rewrote it by using case, which looks even more clear (to me, at least); PTAL.

Reply:
Looks better for me too. Thanks!